File: LLM-Translations/GPT-4/codenet/C++/Go/s422352362.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n int
	fmt.Scan(&n)
	a := make([]int64, n+1)
	pre := make([]int64, n+1)
	last := make([]int64, n+1)

	for i := 1; i <= n; i++ {
		fmt.Scan(&a[i])
	}

	var sumpre, sumlast int64
	for i := 1; i < n; i++ {
		sumpre += a[i]
		pre[i] = sumpre
	}

	for i := n; i > 1; i-- {
		sumlast += a[i]
		last[i] = sumlast
	}

	d := int64(1 << 60)
	for i := 1; i < n; i++ {
		d = int64(math.Min(float64(d), math.Abs(float64(pre[i]-last[i+1]))))
	}

	fmt.Println(d)
}

// 

File: LLM-Translations/GPT-4/codenet/C++/Python/s422352362.py
import sys

def main():
    n = int(input().strip())
    a = list(map(int, input().strip().split()))
    pre = [0] * (n + 1)
    last = [0] * (n + 1)
    sumpre = 0
    sumlast = 0
    
    for i in range(n - 1):
        sumpre += a[i]
        pre[i] = sumpre
        
    for i in range(n - 1, 0, -1):
        sumlast += a[i]
        last[i] = sumlast
        
    d = 1 << 60
    for i in range(n - 1):
        d = min(d, abs(pre[i] - last[i + 1]))
        
    print(d)

if __name__ == "__main__":
    main()


